cmd/terramate/e2etests/core/general_test.go

```diff
@@ -492,6 +491,60 @@ func TestChangedWorksWithoutRemote(t *testing.T) {
 	)
 }
 
+func TestRemoteSafeguardDefaults(t *testing.T) {
+	t.Parallel()
+
+	rootdir := test.TempDir(t)
+
+	git := sandbox.NewGit(t, rootdir)
+	git.InitLocalRepo()
+
+	cli := NewCLI(t, git.BaseDir())
+
+	assertSuccess := func(t *testing.T) {
+		t.Helper()
+
+		AssertRun(t, cli.ListChangedStacks())
+
+		AssertRun(t, cli.Run(
+			"run",
+			"--changed",
+			HelperPath,
+			"cat",
+			"whatever",
+		))
+	}
+
+	path := test.WriteFile(t, git.BaseDir(), "file1", "# generated by terramate")
+	git.Add(path)
+	git.Commit("first commit")
+	path = test.WriteFile(t, git.BaseDir(), "file2", "# generated by terramate")
+	git.Add(path)
+	git.Commit("second commit")
+
+	assertSuccess(t)
```

Comment:
nitpick, we could inline the `assertSuccess` function body and remove that function?

```suggestion
	AssertRun(t, cli.ListChangedStacks())
	AssertRun(t, cli.Run("run", "--changed", HelperPath, "cat", "whatever"))
```